### ISSUE TITLE
Regression in name hashing of traits

### ISSUE THREAD

**Issue by gkossakowski** [sbt/sbt#2436]
``` scala
// A.scala
trait A {
  //def foo: Int = 12
}

// B.scala
class B(a: A)
```

When you uncomment definition of `foo` `B.scala` is recompiled but it shouldn't. The recompilation of `B.scala` caused by change of a hash of `A` caused by addition of `foo`. This regressed in 40ebc825317ed29a2ab87be07246fcdcfc6d1e3d.

I think it's large enough regression that should be considered as a showstopper for 0.13.10.
Unfortunately, existing scripted tests don't cover this scenario.

//cc @eed3si9n

**Comment by gkossakowski** [comment]
@Duhemm: I assigned you to have a look into this.

**Comment by Duhemm** [comment]
Fixed in #2441

**Comment by dwijnand** [comment]
Btw the reason this isn't closed is because GitHub issues close when the `Fixes #XXXX` commit hits the default branch, which for sbt/sbt is 1.0.x.
